### PR TITLE
[ASoC 2022] Pixiu Metrics Implementation

### PR DESCRIPTION
**Pull request by gongna-au**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #433

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/metrics/collector/api_health.go [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+	"github.com/apache/dubbo-go-pixiu/pkg/metrics/global"
+)
+
+// A collection of many Api states
+type ApiStatsResponse struct {
+	Share    ApiStatShare `json:"share"`
+	ApiStats []ApiStat    `json:"api_stats"`
+}
+
+//The state shared by the Api
```
Comment (baerwang) [inline pkg/metrics/collector/api_health.go]: space

File: pkg/metrics/collector/api_health.go [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+package collector
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+)
```
Comment (AlexStocks) [inline pkg/metrics/collector/api_health.go:27]: 下面一行加上一行空格

**Review by AlexStocks — APPROVED** [review]

File: pkg/metrics/collector/api_health.go (hunk 2 above)
Comment (gongna-au) [reply pkg/metrics/collector/api_health.go:27]: ok

File: pkg/metrics/collector/api_health.go [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+		},
+	}
+}
+
+func (ds *ApiHealth) Describe(ch chan<- *prometheus.Desc) {
+	for _, metric := range ds.apiMetrics {
+		ch <- metric.Desc
+	}
+
+	ch <- ds.up.Desc()
```
Comment (baerwang) [inline pkg/metrics/collector/api_health.go:121]: 多个channel 改成一个channel

File: pkg/metrics/global/global.go [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package global
+
+//Mainly complete the encapsulation of some public class code and the definition of some global keywords.
```
Comment (baerwang) [inline pkg/metrics/global/global.go]: space

**Review by baerwang — APPROVED** [review]

**Review by PhilYue — APPROVED** [review]

**Pull request closed without merging**